apps/mobile/src/utils/analytics.ts

```diff
@@ -16,15 +16,43 @@ const segmentClient = createClient({
 
 segmentClient.add({ plugin: new AppLifecycleEventPlugin() });
 
-const leatherAnalyticsClient = configureAnalyticsClient<SegmentClient>({
+const analyticsClient = configureAnalyticsClient({
   client: segmentClient,
   defaultProperties: {
     platform: 'mobile',
   },
 });
 
-export let analytics: ReturnType<typeof configureAnalyticsClient<SegmentClient>> | undefined =
-  leatherAnalyticsClient;
+let analyticsEnabled = selectAnalyticsPreference(store.getState()) === 'consent-given';
+
+function setupPreferenceSubscription() {
+  return store.subscribe(() => {
+    const analyticsPreference = selectAnalyticsPreference(store.getState());
+
+    if (analyticsPreference === 'rejects-tracking') {
+      void analytics.track('user_setting_updated', { analytics: 'rejects-tracking' });
```

Comment:
If they reject tracking I don’t think we should trigger an event to signify that

Reply:
This isn't an added behavior, it was already there before the refactoring. It's pretty common nad a useful metric to have.

Reply:
I know it was there before, just seems non consensual.

Tracking an event to signify unwillingness to be tracked